File: app/src/main/java/com/example/androidcalculatorwithhistory/dummy/HistoryContent.java
package com.example.androidcalculatorwithhistory.dummy

import java.util.ArrayList
import java.util.HashMap

/**
 * Helper class for providing sample content for user interfaces created by
 * Android template wizards.
 *
 * TODO: Replace all uses of this class before publishing your app.
 */
public class HistoryContent {

    public static final List<HistoryItem> ITEMS = new ArrayList<HistoryItem>();
    /**
     * A map of sample (dummy) items, by ID.
     */
    val ITEM_MAP: MutableMap<String, HistoryItem> = HashMap()

    private val COUNT = 25

    init {
        // Add some sample items.
        for (i in 1..COUNT) {
            addItem(createHistoryItem(i))
        }
    }

    private fun addItem(item: HistoryItem) {
        ITEMS.add(item)
        ITEM_MAP.put(item.id, item)
    }

    private fun createHistoryItem(position: Int): HistoryItem {
        return HistoryItem(position.toString(), "Item " + position, makeDetails(position))
    }

    private fun makeDetails(position: Int): String {
        val builder = StringBuilder()
        builder.append("Details about Item: ").append(position)
        for (i in 0..position - 1) {
            builder.append("\nMore details information here.")
        }
        return builder.toString()
    }

    /**
     * A dummy item representing a piece of content.
     */
    data class HistoryItem(val id: String, val content: String, val details: String) {
        override fun toString(): String = content
    }
}

File: app/src/main/java/com/example/androidcalculatorwithhistory/dummy/HistoryContent.kt
package com.example.androidcalculatorwithhistory.dummy

import java.util.ArrayList
import org.joda.time.DateTime


/**
 * Helper class for providing sample content for user interfaces created by
 * Android template wizards.
 *
 * TODO: Replace all uses of this class before publishing your app.
 */
object HistoryContent {

    /**
     * An array of sample (dummy) items.
     */
    val ITEMS: MutableList<HistoryItem> = ArrayList()

    init {
        val now: DateTime = DateTime.now()
        addItem(HistoryItem(2.0, 1.829, "Length", "Yards", "Meters", now.minusDays(1).toString()))
        addItem(HistoryItem(1.0, 3.785, "Volume", "Gallons", "Liters", now.minusDays(1).toString()))
        addItem(HistoryItem(2.0, 1.829, "Length", "Yards", "Meters", now.plusDays(1).toString()))
        addItem(HistoryItem(1.0, 3.785, "Volume", "Gallons", "Liters", now.plusDays(1).toString()))
    }


    public fun addItem(item: HistoryItem) {
        ITEMS.add(item)
    }

    private fun makeDetails(position: Int): String {
        val builder = StringBuilder()
        builder.append("Details about Item: ").append(position)
        for (i in 0..position - 1) {
            builder.append("\nMore details information here.")
        }
        return builder.toString()
    }

    /**
     * A dummy item representing a piece of content.
     */
    class HistoryItem(
        val fromVal: Double? = 0.0,
        val toVal: Double? = 0.0,
        val mode: String = "Length",
        val fromUnits: String = "Miles",
        val toUnits: String = "Meters",
        val timestamp: String = DateTime.now().toString(),
        var _key: String = "None"
    ) {


        override fun toString(): String {
            return this.fromVal.toString() + " " + this.fromUnits + " = " + this.toVal + " " + this.toUnits
        }
    }

}
